spark/src/test/scala/org/apache/spark/sql/CometTestBase.scala

```diff
@@ -377,6 +382,36 @@ abstract class CometTestBase
         .write
         .option("parquet.enable.dictionary", withDictionary.toString)
         .parquet(file.getCanonicalPath)
+
+      if (withDictionary) {
+        // if the test specified to write dictionary-encoded data, we should check that we actually wrote some
+        // dictionary-encoded data
+        val files = file.listFiles(new FilenameFilter {
```

Comment:
a nit: I feel this code checks if any column has RLE encoding, however RLE is on row group level? that means every page of the every column may or may not encoded. If lets say column C1 encoded, C2 nope, than no warning shown but we use exactly C2 for testing